queries/rescript/highlights.scm

```diff
@@ -14,23 +14,34 @@
   (list_pattern)
 ] @type
 
+((type_identifier) @type.builtin
+  (#any-of? @type.builtin
+    "int" "char" "string" "float" "bool" "unit"))
+
 [
   (variant_identifier)
   (polyvar_identifier)
-] @constant
+] @constructor
```

Comment:
Not everyone might agree on this one, since variants, specially polyvars without parameters, are used as "constants". This thinking is common when making JS bindings, since polyvars compile to strings.

However, from a language perspective, I find it more correct to think of them as constructors. This is also how the ocaml parser highlights them.